pyexchange/etoro.py

```diff
@@ -0,0 +1,332 @@
+# This file is part of Maker Keeper Framework.
+#
+# Copyright (C) 2017-2019 MikeHathaway 
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+from pprint import pformat
+from typing import Optional, List
+
+import dateutil.parser
+import requests
+import time
+import hmac
+import hashlib
+
+import uuid
+import base64
+
+from urllib.parse import urlencode
+
+from pyexchange.api import PyexAPI
+
+from pymaker.numeric import Wad
+from pymaker.util import http_response_summary
+
+from Crypto.Signature import pkcs1_15
+from Crypto.Hash import SHA256
+from Crypto.PublicKey import RSA
+
+class Order:
+    def __init__(self,
+                 order_id: str,
+                 timestamp: int, # time in seconds
+                 instrument_id: str,
+                 is_sell: bool,
+                 price: Wad,
+                 amount: Wad,
+                 remaining_amount: Wad):
+
+        assert(isinstance(instrument_id, str))
+        assert(isinstance(timestamp, int))
+        assert(isinstance(is_sell, bool))
+        assert(isinstance(price, Wad))
+        assert(isinstance(amount, Wad))
+        assert(isinstance(remaining_amount, Wad))
+
+        self.order_id = order_id
+        self.timestamp = timestamp
+        self.instrument_id = instrument_id
+        self.is_sell = is_sell
+        self.price = price
+        self.amount = amount
+        self.remaining_amount = remaining_amount
+
+    @property
+    def sell_to_buy_price(self) -> Wad:
+        return self.price
+
+    @property
+    def buy_to_sell_price(self) -> Wad:
+        return self.price
+
+    @property
+    def remaining_buy_amount(self) -> Wad:
+        return self.remaining_amount*self.price if self.is_sell else self.remaining_amount
+
+    @property
+    def remaining_sell_amount(self) -> Wad:
+        return self.remaining_amount if self.is_sell else self.remaining_amount*self.price
+
+    def __repr__(self):
+        return pformat(vars(self))
+
+    @staticmethod
+    def to_order(item):
```

Comment:
Since this factory method sits in the `Order` class, I'd expect a name like `from_message`.  Current name suggests we're trying to convert an `Order` to another object.

Reply:
Makes sense. Updated name to from_message